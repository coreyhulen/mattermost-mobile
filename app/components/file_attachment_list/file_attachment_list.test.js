// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import React from 'react';
import {shallow} from 'enzyme';

import FileAttachment from './file_attachment_list.js';
import Preferences from '@mm-redux/constants/preferences';

jest.mock('react-native-file-viewer', () => ({
    open: jest.fn(),
}));

describe('FileAttachmentList', () => {
    const files = [{
        create_at: 1546893090093,
        delete_at: 0,
        extension: 'png',
        has_preview_image: true,
        height: 171,
        id: 'fileId',
        mime_type: 'image/png',
        name: 'image01.png',
        post_id: 'postId',
        size: 14894,
        update_at: 1546893090093,
        user_id: 'userId',
        width: 425,
    },
    {
        create_at: 1546893090093,
        delete_at: 0,
        extension: 'png',
        has_preview_image: true,
        height: 800,
        id: 'otherFileId',
        mime_type: 'image/png',
        name: 'image02.png',
        post_id: 'postId',
        size: 24894,
        update_at: 1546893090093,
        user_id: 'userId',
        width: 555,
    }];

    const nonImage = {
        extension: 'other',
        id: 'fileId',
        mime_type: 'other/type',
        name: 'file01.other',
        post_id: 'postId',
        size: 14894,
        user_id: 'userId',
    };

    const baseProps = {
        canDownloadFiles: true,
        deviceHeight: 680,
        deviceWidth: 660,
        fileIds: ['fileId'],
        files: [files[0]],
        postId: 'postId',
        theme: Preferences.THEMES.default,
    };

    test('should match snapshot with a single image file', () => {
        const wrapper = shallow(
            <FileAttachment {...baseProps}/>,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot with two image files', () => {
        const props = {
            ...baseProps,
            files,
        };

        const wrapper = shallow(
            <FileAttachment {...props}/>,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot with three image files', () => {
        const thirdImage = {...files[1], id: 'thirdFileId', name: 'image03.png'};
        const props = {
            ...baseProps,
            files: [...files, thirdImage],
        };

        const wrapper = shallow(
            <FileAttachment {...props}/>,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot with four image files', () => {
        const thirdImage = {...files[1], id: 'thirdFileId', name: 'image03.png'};
        const fourthImage = {...files[1], id: 'fourthFileId', name: 'image04.png'};

        const props = {
            ...baseProps,
            files: [...files, thirdImage, fourthImage],
        };

        const wrapper = shallow(
            <FileAttachment {...props}/>,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot with more than four image files', () => {
        const thirdImage = {...files[1], id: 'thirdFileId', name: 'image03.png'};
        const fourthImage = {...files[1], id: 'fourthFileId', name: 'image04.png'};
        const fifthImage = {...files[1], id: 'fifthFileId', name: 'image05.png'};
        const sixthImage = {...files[1], id: 'sixthFileId', name: 'image06.png'};

        const props = {
            ...baseProps,
            files: [...files, thirdImage, fourthImage, fifthImage, sixthImage],
        };

        const wrapper = shallow(
            <FileAttachment {...props}/>,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot with non-image attachment', () => {
        const props = {
            ...baseProps,
            files: [nonImage],
        };

        const wrapper = shallow(
            <FileAttachment {...props}/>,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot with combination of image and non-image attachments', () => {
        const props = {
            ...baseProps,
            files: [...files, nonImage],
        };

        const wrapper = shallow(
            <FileAttachment {...props}/>,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should call getFilesForGallery on props change', async () => {
        const props = {
            ...baseProps,
        };

        const wrapper = shallow(
            <FileAttachment {...props}/>,
        );

        wrapper.instance().getFilesForGallery = jest.fn().mockImplementationOnce(() => []);
        wrapper.setProps({files: [files[0], files[1]]});
        expect(wrapper.instance().getFilesForGallery).toHaveBeenCalled();
    });
});
