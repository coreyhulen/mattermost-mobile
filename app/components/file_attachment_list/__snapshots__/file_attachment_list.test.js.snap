// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileAttachmentList should match snapshot with a single image file 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "row",
        },
        Object {
          "width": 657,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <FileAttachment
        canDownloadFiles={true}
        file={
          Object {
            "caption": "image01.png",
            "data": Object {
              "create_at": 1546893090093,
              "delete_at": 0,
              "extension": "png",
              "has_preview_image": true,
              "height": 171,
              "id": "fileId",
              "mime_type": "image/png",
              "name": "image01.png",
              "post_id": "postId",
              "size": 14894,
              "update_at": 1546893090093,
              "user_id": "userId",
              "width": 425,
            },
          }
        }
        id="fileId"
        index={0}
        isSingleImage={true}
        onCaptureRef={[Function]}
        onPreviewPress={[Function]}
        theme={
          Object {
            "awayIndicator": "#ffbc42",
            "buttonBg": "#166de0",
            "buttonColor": "#ffffff",
            "centerChannelBg": "#ffffff",
            "centerChannelColor": "#3d3c40",
            "codeTheme": "github",
            "dndIndicator": "#f74343",
            "errorTextColor": "#fd5960",
            "linkColor": "#2389d7",
            "mentionBg": "#ffffff",
            "mentionBj": "#ffffff",
            "mentionColor": "#145dbf",
            "mentionHighlightBg": "#ffe577",
            "mentionHighlightLink": "#166de0",
            "newMessageSeparator": "#ff8800",
            "onlineIndicator": "#06d6a0",
            "sidebarBg": "#145dbf",
            "sidebarHeaderBg": "#1153ab",
            "sidebarHeaderTextColor": "#ffffff",
            "sidebarText": "#ffffff",
            "sidebarTextActiveBorder": "#579eff",
            "sidebarTextActiveColor": "#ffffff",
            "sidebarTextHoverBg": "#4578bf",
            "sidebarUnreadText": "#ffffff",
            "type": "Mattermost",
          }
        }
        wrapperWidth={657}
      />
    </View>
  </View>
</View>
`;

exports[`FileAttachmentList should match snapshot with multiple image files 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "row",
        },
        Object {
          "width": 657,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <FileAttachment
        canDownloadFiles={true}
        file={
          Object {
            "caption": "image01.png",
            "data": Object {
              "create_at": 1546893090093,
              "delete_at": 0,
              "extension": "png",
              "has_preview_image": true,
              "height": 171,
              "id": "fileId",
              "mime_type": "image/png",
              "name": "image01.png",
              "post_id": "postId",
              "size": 14894,
              "update_at": 1546893090093,
              "user_id": "userId",
              "width": 425,
            },
          }
        }
        id="fileId"
        index={0}
        isSingleImage={false}
        onCaptureRef={[Function]}
        onPreviewPress={[Function]}
        theme={
          Object {
            "awayIndicator": "#ffbc42",
            "buttonBg": "#166de0",
            "buttonColor": "#ffffff",
            "centerChannelBg": "#ffffff",
            "centerChannelColor": "#3d3c40",
            "codeTheme": "github",
            "dndIndicator": "#f74343",
            "errorTextColor": "#fd5960",
            "linkColor": "#2389d7",
            "mentionBg": "#ffffff",
            "mentionBj": "#ffffff",
            "mentionColor": "#145dbf",
            "mentionHighlightBg": "#ffe577",
            "mentionHighlightLink": "#166de0",
            "newMessageSeparator": "#ff8800",
            "onlineIndicator": "#06d6a0",
            "sidebarBg": "#145dbf",
            "sidebarHeaderBg": "#1153ab",
            "sidebarHeaderTextColor": "#ffffff",
            "sidebarText": "#ffffff",
            "sidebarTextActiveBorder": "#579eff",
            "sidebarTextActiveColor": "#ffffff",
            "sidebarTextHoverBg": "#4578bf",
            "sidebarUnreadText": "#ffffff",
            "type": "Mattermost",
          }
        }
        wrapperWidth={657}
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <FileAttachment
        canDownloadFiles={true}
        file={
          Object {
            "caption": "image02.png",
            "data": Object {
              "create_at": 1546893090093,
              "delete_at": 0,
              "extension": "png",
              "has_preview_image": true,
              "height": 800,
              "id": "otherFileId",
              "mime_type": "image/png",
              "name": "image02.png",
              "post_id": "postId",
              "size": 24894,
              "update_at": 1546893090093,
              "user_id": "userId",
              "width": 555,
            },
          }
        }
        id="otherFileId"
        index={1}
        isSingleImage={false}
        onCaptureRef={[Function]}
        onPreviewPress={[Function]}
        theme={
          Object {
            "awayIndicator": "#ffbc42",
            "buttonBg": "#166de0",
            "buttonColor": "#ffffff",
            "centerChannelBg": "#ffffff",
            "centerChannelColor": "#3d3c40",
            "codeTheme": "github",
            "dndIndicator": "#f74343",
            "errorTextColor": "#fd5960",
            "linkColor": "#2389d7",
            "mentionBg": "#ffffff",
            "mentionBj": "#ffffff",
            "mentionColor": "#145dbf",
            "mentionHighlightBg": "#ffe577",
            "mentionHighlightLink": "#166de0",
            "newMessageSeparator": "#ff8800",
            "onlineIndicator": "#06d6a0",
            "sidebarBg": "#145dbf",
            "sidebarHeaderBg": "#1153ab",
            "sidebarHeaderTextColor": "#ffffff",
            "sidebarText": "#ffffff",
            "sidebarTextActiveBorder": "#579eff",
            "sidebarTextActiveColor": "#ffffff",
            "sidebarTextHoverBg": "#4578bf",
            "sidebarUnreadText": "#ffffff",
            "type": "Mattermost",
          }
        }
        wrapperWidth={657}
      />
    </View>
  </View>
</View>
`;
